### PR TITLE
Add volume support

### PR DESCRIPTION
**Pull request by WeiZhang555**
Currently runv don't support docker volumes(OCI Mounts fields), this
commit will add volume support, so that we can bind mount host dir or make a
volume dir in container.

Signed-off-by: Zhang Wei <zhangwei555@huawei.com>

### CODE REVIEW

File: supervisor/container.go [hunk 1]
```diff
@@ -252,3 +263,130 @@ func (c *Container) reap() {
...
+		}
+		if err := mount.Mount(m.Source, dest, m.Type, strings.Join(m.Options, ",")); err != nil {
+			return err
+		}
+		if stat != nil {
+			if err = os.Chmod(dest, stat.Mode()); err != nil {
+				return err
+			}
+		}
+		return nil
```
Comment (laijs) [inline supervisor/container.go]: proc sysfs mqueue tmpfs MUST pass into hyperstart and let hyperstart mount it.
hyperstart haven't has such abi, so they should be skipped.
Could you add a issue for such purpose?

File: supervisor/container.go [hunk 2]
```diff
@@ -8,8 +8,11 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strings"
 	"time"
 
+	"github.com/docker/docker/pkg/mount"
+	"github.com/docker/docker/pkg/symlink"
```
Comment (laijs) [inline supervisor/container.go]: please check the travis error

supervisor/container.go:14:2: cannot find package "github.com/docker/docker/pkg/mount"

File: supervisor/container.go (hunk 1 above)
Comment (WeiZhang555) [inline supervisor/container.go]: OK. I'll skip it.

File: supervisor/container.go [hunk 3]
```diff
@@ -252,3 +263,130 @@ func (c *Container) reap() {
...
+// checkMountDestination checks to ensure that the mount destination is not over the top of /proc.
+// dest is required to be an abs path and have any symlinks resolved before calling this function.
+func checkMountDestination(rootfs, dest string) error {
+	invalidDestinations := []string{
+		"/proc",
+	}
+	// White list, it should be sub directories of invalid destinations
+	validDestinations := []string{
+		// These entries can be bind mounted by files emulated by fuse,
+		// so commands like top, free displays stats in container.
```
Comment (gao-feng) [inline supervisor/container.go]: seams like runv doesn't need to emulate these files.

Comment (WeiZhang555) [inline supervisor/container.go]: It looks like that mount points in white list can be used by `fuse` for better isolation. I'm not sure how can runv make use of this, but maybe we can leave it here. 
It wont harm anyone, right?  If not, I can move mount points in white list(validDestinations ) into black list(invalidDestinations)

**Comment by WeiZhang555** [comment]
Updated. Test is green now. :smile:

**Comment by laijs** [comment]
related: https://github.com/hyperhq/hyperd/issues/285

**Pull request merged**